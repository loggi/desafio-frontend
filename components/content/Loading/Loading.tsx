import Alert from '@mui/material/Alert';

export const LoadingMessage = () => {
  return (
      <Alert sx={{ width: '100%' }} severity="info">Estamos buscando os dados. Por favor aguarde.</Alert>
  );
}