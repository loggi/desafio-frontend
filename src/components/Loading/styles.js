import styled from '@emotion/styled'
import { css } from '@emotion/react'

const WrapperModifier = {
  visibleWrapper: () => css`
    position: absolute;
    top: 0;
    bottom: 0;
    z-index: 99;
    width: 100%;
  `,
  visibleImg: () => css`
    z-index: 999;
  `
}

export const Wrapper = styled.div`
  ${({ theme, visible }) => css`
    display: flex;
    align-items: center;
    justify-content: center;

    ${visible && WrapperModifier.visibleWrapper(theme)};

    img {
      width: 64px;
      height: 64px;

      ${visible && WrapperModifier.visibleImg(theme)};
    }
  `}
`

export const Backdrop = styled.div`
  position: absolute;
  top: 0;
  width: 100%;
  height: 100%;
`
