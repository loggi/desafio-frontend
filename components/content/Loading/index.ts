import { LoadingMessage } from "./Loading";
export { LoadingMessage }