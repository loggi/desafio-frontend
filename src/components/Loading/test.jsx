import { render, screen } from '@testing-library/react'

import Loading from '.'

describe('<Loading />', () => {
  it('should render the Loading', () => {
    const { container } = render(<Loading visible />)

    expect(screen.getByRole('img', { name: /Carregando/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
