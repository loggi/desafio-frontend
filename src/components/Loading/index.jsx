import PropTypes from 'prop-types'
import Image from 'next/image'

import * as S from './styles'

const Loading = ({ visible }) => (
  <S.Wrapper visible={visible}>
    {visible && <S.Backdrop />}
    <Image
      src="/assets/icons/spinner.svg"
      alt="Carregando"
      width={32}
      height={32}
    />
  </S.Wrapper>
)

Loading.propTypes = {
  visible: PropTypes.bool
}

export default Loading
