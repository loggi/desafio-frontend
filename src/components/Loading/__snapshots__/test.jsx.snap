// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should render the Loading 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: absolute;
  top: 0;
  bottom: 0;
  z-index: 99;
  width: 100%;
}

.emotion-0 img {
  width: 64px;
  height: 64px;
  z-index: 999;
}

.emotion-1 {
  position: absolute;
  top: 0;
  width: 100%;
  height: 100%;
}

<div
  class="emotion-0"
>
  <div
    class="emotion-1"
  />
  <img
    alt="Carregando"
    data-nimg="1"
    decoding="async"
    height="32"
    loading="lazy"
    src="/assets/icons/spinner.svg"
    style="color: transparent;"
    width="32"
  />
</div>
`;
